src/tequila/grouping/overlapping_methods.py

```diff
@@ -0,0 +1,203 @@
+    OverlappingGroups. Eventually, this may also be used for building cov_dict from
+    approximate wavefunction if the user provides only the latter.
+    '''
+    def __init__(self, cov_dict, n_iter=5):
+        self.cov_dict = cov_dict
+        self.n_iter = n_iter
+
+class OverlappingGroups_wo_fixed:
+    '''
+    Class required for mapping OverlappingGroups onto a vector of overlapping coefficients.
+    Same as OverlappingGroups, but Pauli products corresponding to fixed coefficients are removed.
+    '''
+    def __init__(self, o_groups, o_terms, term_exists_in):
+        def exclude_fixed_coeffs(o_groups, o_terms, term_exists_in):
+            '''
+            Remove fixed coefficients from o_groups and term_exists_in.
+            These fixed coefficients are determined from other Pauli coefficients in order that
+            the condition sum_alpha c^{alpha} = 0 is satisfied.
+            '''
+            fixed_grp = []
+            o_groups_wo_fixed = deepcopy(o_groups)
+            term_exists_in_wo_fixed = deepcopy(term_exists_in)
+            for term_idx, term in enumerate(o_terms):
+                fixed_grp.append(term_exists_in[term_idx][-1])
+                o_groups_wo_fixed[fixed_grp[term_idx]].remove(term)
+                term_exists_in_wo_fixed[term_idx].remove(fixed_grp[term_idx])
+            Ncoeff_grp = np.array([len(lst) for lst in o_groups_wo_fixed])
+            init_idx = [sum(Ncoeff_grp[:i]) for i in range(len(o_groups_wo_fixed))]
+            return fixed_grp, o_groups_wo_fixed, term_exists_in_wo_fixed, Ncoeff_grp, init_idx
+
+        def get_term_idxs(o_terms, o_groups_wo_fixed, term_exists_in_wo_fixed, init_idx):
+            '''
+            Obtain a dictionary (over i) of dictionaries (over alpha) containing
+            the index corresponding to a Pauli coefficient c_{i}^{alpha}.
+            '''
+            term_idxs = {}
+            for term_idx, term in enumerate(o_terms):
+                cur_idxs = {}
+                for grp_idx in term_exists_in_wo_fixed[term_idx]:
+                    cur_idxs[grp_idx] = init_idx[grp_idx] + o_groups_wo_fixed[grp_idx].index(term)
+                term_idxs[term.binary_tuple()] = cur_idxs
+            return term_idxs
+
+        self.fixed_grp, self.o_groups, self.term_exists_in, self.Ncoeff_grp, self.init_idx \
+                                                            = exclude_fixed_coeffs(o_groups, o_terms, term_exists_in)
+        self.term_idxs = get_term_idxs(o_terms, self.o_groups, self.term_exists_in, self.init_idx)
+        return None
+
+def get_cov(term1, term2, cov_dict):
+    '''
+    Return the covariance between Pauli string term 1 and 2 from a dictionary.
+    '''
+    if (term1.binary_tuple(), term2.binary_tuple()) in cov_dict:
+        return cov_dict[(term1.binary_tuple(), term2.binary_tuple())]
+    if (term2.binary_tuple(), term1.binary_tuple()) in cov_dict:
+        return cov_dict[(term2.binary_tuple(), term1.binary_tuple())]
+
```

Comment:
This assumes that `term1` and `term2` commute. This assumption should be in docstring.